### PR TITLE
CAD-660 darwin counters

### PR DESCRIPTION
**Pull request by CodiePP**
description
-----------

- [x] counters on MacOSX/Darwin

output on a Mac:

```
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.bytesAllocated = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.liveBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.maxLiveBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.maxLargeBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.maxCompactBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.maxSlopBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.maxUsedMemBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.gcLiveBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.gcCopiedBytes = 0 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.gcCpuNs = 0.0 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.gcElapsedNs = 0.0 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.cpuNs = 2.2e-4 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.elapsedNs = 3.6572e-4 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.gcNum = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] RTS.gcMajorNum = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] IO.ndisks = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.interfaces = 10
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifn_"summary" = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-mtu = 30264 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-ipackets = 4766773
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-ierrors = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-opackets = 1807907
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-oerrors = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-collisions = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-ibytes = 959728640 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-obytes = 368784384 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-imcast = 641191
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Net.ifd_0-omcast = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Stat.Pid = 32808
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Stat.user_time = 1.5352714e10 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Stat.system_time = 4.9055857e9 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Stat.policy = 1
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Stat.suspend_count = 0
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Mem.Pid = 32808
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Mem.virtual_size = 1103910191104 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Mem.resident_size = 5664768 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Mem.resident_size_max = 5664768 B
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Sys.boot_time = 1588347521 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Sys.SysUserTime = 4.332867 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Sys.SystemTime = 17.283087 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Sys.IdleTime = 10.040732 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Sys.NiceTime = 0.0 s
[counters.node-metrics:Notice:11] [2020-05-07 11:17:20.93 UTC] Sys.Platform = 2
```

checklist
---------

- [x] compiles (`cabal v2-build` or `stack build`)
- [x] tests run successfully (`cabal v2-test` or `stack test`)
- [ ] documentation added
- [x] link to an issue
- [ ] add milestone (the current sprint)

### CODE REVIEW

**Comment by CodiePP** [comment]
commented out the part that required to link to native OSX frameworks.
this also means that we do not get information about disk I/O.

**Comment by CodiePP** [comment]
further trim down to reduce the risk of leaking space in C-allocated memory. verified in profiling "example-complex" and resident memory seems to be stable; all fine.

File: iohk-monitoring/iohk-monitoring.cabal [hunk 1]
```diff
@@ -130,7 +133,7 @@ library
   if flag(performance-test-queue)
     cpp-options:       -DPERFORMANCE_TEST_QUEUE
 
-  ghc-options:         -Wall -Werror
```
Comment (denisshevchenko) [inline iohk-monitoring/iohk-monitoring.cabal:133]: Why did you remove `-Werror`? It helps us to keep the code as clean as possible, no?

Comment (CodiePP) [reply iohk-monitoring/iohk-monitoring.cabal:133]: oh, I should readd it.
